File: examples_py/helloworld/publisher.py
import zv_robot_sdk_python as zv
import time

print(f"zv_robot_sdk_python version: {zv.__version__}")

HELLO_WORLD_TOPIC = "rt/hello_world"

zv.ChannelFactory.Instance().Init(0)
publisher = zv.ChannelPublisher_String_(HELLO_WORLD_TOPIC)
publisher.InitChannel()

count = 0
while True:
    msg = zv.String_(f"HelloWorld. {count}")
    count += 1
    publisher.Write(msg)
    time.sleep(1)

File: examples/helloworld/publisher.cpp
#include <thread>

#include <zv/robot/channel/channel_publisher.hpp>
#include "String_.hpp"

static const std::string HELLO_WORLD_TOPIC = "rt/hello_world";

using namespace zv::robot;
using namespace zv::common;

int main() {
    ChannelFactory::Instance()->Init(0);
    ChannelPublisher<std_msgs::msg::dds_::String_> publisher(HELLO_WORLD_TOPIC);

    publisher.InitChannel();

    int count = 0;
    while (true)
    {
        std_msgs::msg::dds_::String_ msg("HelloWorld. "  + std::to_string(count++));
        publisher.Write(msg);
        std::this_thread::sleep_for(std::chrono::seconds(1));
    }

    return 0;
}
